docker/worker/worker.py

```diff
@@ -476,6 +491,12 @@ def _do_update(self, source_repo, repo, vulnerability, relative_path,
       logging.warning('%s has an encoding error, skipping.', vulnerability.id)
       return
 
+    # If the vulnerability has affected packages, but none of them are supported
+    # then discard it
+    if len(vulnerability.affected) > 0:
+      if filter_unsupported_ecosystems(vulnerability) == 0:
```

Comment:
I think we still want to update existing bugs if affected packages changes.

e.g. if a package has 2 affected packages, one for an ecosystem we support, one we don't. If they then remove the one that affects the ecosystem we support, we would want to update that in the database. 

I don't think we need to filter out the vulnerability at all, just import them and set status to invalid if there are no affected packages. This way importer won't try to keep importing filtered out vulns for unsupported ecosystems.

Reply:
+1 good point. There's a case here of an existing entry getting updated to be invalid later with no valid ecosystenms. 

We can just import everything as Rex suggested, and simply mark them as invalid in these cases (similar to https://github.com/google/osv.dev/blob/7ccaa3a3bba95364039bac82c8dd4bb48d952d1b/docker/worker/worker.py#L419).